exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestNewSimpleRepeatedFunctions.java

```diff
@@ -24,154 +24,72 @@
-        .unOrdered()
-        .baselineColumns("EXPR$0")
-        .baselineValuesForSingleColumn(2, 3, 0, 1, 5)
-        .go();
+  @Test
+  public void testRepeatedCountRepeatedMapInWhere() throws Exception {
+    String query = SELECT_REPEATED_COUNT_MAP + " where repeated_count(mapArray) > 2";
+    performTest(query, 3, 5);
+  }
 
-    testBuilder()
-        .sqlQuery(selectQuery + " group by 1 having repeated_count(mapArray) < 3", fileName)
-        .unOrdered()
-        .baselineColumns("EXPR$0")
-        .baselineValuesForSingleColumn(2, 0, 1)
-        .go();
+  @Test
+  public void testRepeatedCountRepeatedMapInHaving() throws Exception {
+    String query = SELECT_REPEATED_COUNT_MAP + " group by 1 having repeated_count(mapArray) < 3";
+    performTest(query, 2, 0, 1);
   }
 
   @Test
   public void testRepeatedCountRepeatedList() throws Exception {
-    // Contents of the generated file:
-    /*
-      {"id": 1, "array": [[1, 2], [1, 3], [2, 3]]}
-      {"id": 2, "array": []}
-      {"id": 3, "array": [[2, 3], [1, 3, 4]]}
-      {"id": 4, "array": [[1], [2], [3, 4], [5], [6]]}
-      {"id": 5, "array": [[1, 2, 3], [4, 5], [6], [7], [8, 9], [2, 3], [2, 3], [2, 3], [2]]}
-      {"id": 6, "array": [[1, 2], [3], [4], [5]]}
-      {"id": 7, "array": []}
-      {"id": 8, "array": [[1], [2], [3]]}
-    */
-    String fileName = "repeated_count_list.json";
-    try (BufferedWriter writer = new BufferedWriter(new FileWriter(
-        new File(dirTestWatcher.getRootDir(), fileName)))) {
-      String[] arrayElements = {
-          "[1, 2], [1, 3], [2, 3]",
-          "",
-          "[2, 3], [1, 3, 4]",
-          "[1], [2], [3, 4], [5], [6]",
-          "[1, 2, 3], [4, 5], [6], [7], [8, 9], [2, 3], [2, 3], [2, 3], [2]",
-          "[1, 2], [3], [4], [5]",
-          "",
-          "[1], [2], [3]"};
-      int elementId = 1;
-      for (String value : arrayElements) {
-        String entry = String.format("{\"id\": %d, \"array\": [%s]}\n", elementId++, value);
-        writer.write(entry);
-      }
-    }
+    performTest(SELECT_REPEATED_COUNT_LIST, 3, 0, 2, 5, 9, 4, 0, 3);
+  }
 
-    String selectQuery = "select repeated_count(array) from dfs.`%s`";
-    testBuilder()
-        .sqlQuery(selectQuery, fileName)
-        .ordered()
-        .baselineColumns("EXPR$0")
-        .baselineValuesForSingleColumn(3, 0, 2, 5, 9, 4, 0, 3)
-        .go();
+  @Test
+  public void testRepeatedCountRepeatedListInWhere() throws Exception {
+    String query = SELECT_REPEATED_COUNT_LIST + " where repeated_count(array) > 4";
+    performTest(query, 5, 9);
+  }
 
-    testBuilder()
-        .sqlQuery(selectQuery + " where repeated_count(array) > 4", fileName)
-        .ordered()
-        .baselineColumns("EXPR$0")
-        .baselineValuesForSingleColumn(5, 9)
-        .go();
+  @Test
+  public void testRepeatedCountRepeatedListInHaving() throws Exception {
+    String query = SELECT_REPEATED_COUNT_LIST + " group by 1 having repeated_count(array) < 4";
```

Comment:
There is overloaded `sqlQuery(String query, Object... replacements)` method.

Reply:
Not sure why this should be used - I find the concatenation fine. If you have any objections, please feel free to point it out.
Just to be sure, do you mean to have the base query with `%s` so the additional part should be formatted (well, passed to the method as the second parameter)?

Reply:
agree, it will be more complicated with `String.format`